### ISSUE TITLE
[a11y]: SR is not announcing the activated link/button or activation of a new tab

### ISSUE THREAD

**Issue by rileydanejohnston** [carbon-design-system/carbon#19145]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.77.0

### React version

16.14.0

### Automated testing tool and ruleset

IBM equal accessibility checker

### Assistive technology

JAWS, Voiceover

### Description

Original issue: https://github.ibm.com/ibmcloud/billing/issues/3384

SR is not announcing the activated link, and when users click on the "Contact Sales" link, it redirects them to another tab.

Recommended fix: SR must announe that new page activates in new tab after clicking on "Contact Sales" link

User impact: Screen reader users experience confusion as the "Contact Sales" link opens in a new tab without any announcement.

WCAG Mapping: 3.2.2 On Input
Level of Compliance: A
Severity: Medium
Test tool: JAWS Firefox
Screen print:

![Image](https://github.com/user-attachments/assets/8e43da0d-b0e2-4c89-9aeb-eb399c99b1d9)

### WCAG 2.1 Violation

WCAG Mapping: 3.2.2 On Input

### Reproduction/example

https://stackblitz.com/edit/github-5suztuzv?file=src%2FApp.jsx

### Steps to reproduce

1. go to https://test.cloud.ibm.com/billing
2. navigate to the "contact sales" button via screen reader
3. click contact sales button
4. screen reader does not announce new tab

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
Thanks for opening this up! If a button with href is going to open in a new tab/window, you'll need to provide that context by adding an icon. From the [current guidance](https://carbondesignsystem.com/components/button/usage/#:~:text=The%20launch%20icon%20should%20be%20used%20on%20any%20call%20to%20action%20that%20launches%20the%20user%20into%20another%20tab%20(whether%20the%20content%20of%20the%20new%20tab%20is%20part%20of%20the%20same%20product%20or%20an%20entirely%20separate%20web%20resource)) on the Carbon website:

> The launch icon should be used on any call to action that launches the user into another tab (whether the content of the new tab is part of the same product or an entirely separate web resource).
>
> ... The target destination of the launch action should be made clear to the user through the button or link label and the surrounding context.

It's a good approach because both sighted and screenreader users benefit from an icon like this.

---

Some additional context that I dug up when looking into this:

* [When to use target=”_blank”](https://css-tricks.com/use-target_blank/), it's generally not best practice.
* NVDA is quite explicit that this is not the responsibility of the screenreader for a number of reasons, https://github.com/nvaccess/nvda/issues/14875#issuecomment-2005168601
* Similarly for JAWS, the only specced behavior is to [announce same page links](https://freedomscientific.github.io/standards-support/html.html#:~:text=Same%20Page%20Link%3A%20This%20indicates%20a%20link%20that%20moves%20you%20to%20another%20location%20on%20the%20same%20page.). It does not mention links opened in a new tab/window.
